src/React/assets/dist/render_controller.js

```diff
@@ -1,83 +1,47 @@
-import { Controller } from '@hotwired/stimulus';
-import React from 'react';
-import require$$0 from 'react-dom';
-
-var client = {};
-
-var hasRequiredClient;
-
-function requireClient () {
-	if (hasRequiredClient) return client;
-	hasRequiredClient = 1;
-
-	var m = require$$0;
-	if (process.env.NODE_ENV === 'production') {
-	  client.createRoot = m.createRoot;
-	  client.hydrateRoot = m.hydrateRoot;
-	} else {
-	  var i = m.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
-	  client.createRoot = function(c, o) {
-	    i.usingClientEntryPoint = true;
-	    try {
-	      return m.createRoot(c, o);
-	    } finally {
-	      i.usingClientEntryPoint = false;
-	    }
-	  };
-	  client.hydrateRoot = function(c, h, o) {
-	    i.usingClientEntryPoint = true;
-	    try {
-	      return m.hydrateRoot(c, h, o);
-	    } finally {
-	      i.usingClientEntryPoint = false;
-	    }
-	  };
-	}
-	return client;
-}
```

Comment:
This is the code from react-dom about handling things differently between development and prod envs. It requires `react-dom` and not `react-dom/client`, as explained in previous PRs.

Now, only the production path is handled, ESBuild does not need anymore to import `react-dom` with the additional logic.